pallets/staking/rpc/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+use jsonrpc_core::{Error as RpcError, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use pallet_staking_rpc_runtime_api::StakingApi as StakingRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{
+    generic::BlockId,
+    traits::{Block as BlockT, ProvideRuntimeApi},
+    Perbill,
+};
+use std::sync::Arc;
+
+#[rpc]
+pub trait StakingApi<BlockHash> {
+    #[rpc(name = "staking_getCurve")]
+    fn get_curve(&self, at: Option<BlockHash>) -> Result<Vec<(Perbill, Perbill)>>;
+}
+
+/// A struct that implements the [`StakingApi`].
+pub struct Staking<C, M> {
+    // If you have more generics, no need to Staking<C, M, N, P, ...>
+    // just use a tuple like Staking<C, (M, N, P, ...)>
+    client: Arc<C>,
+    _marker: std::marker::PhantomData<M>,
+}
+
+impl<C, M> Staking<C, M> {
+    /// Create new `Staking` instance with the given reference to the client.
+    pub fn new(client: Arc<C>) -> Self {
+        Self {
+            client,
+            _marker: Default::default(),
+        }
+    }
+}
+
+/// Error type of this RPC api.
+pub enum Error {
+    /// The transaction was not decodable.
+    DecodeError,
+    /// The call to runtime failed.
+    RuntimeError,
+}
+
+impl From<Error> for i64 {
+    fn from(e: Error) -> i64 {
+        match e {
+            Error::RuntimeError => 1,
+            Error::DecodeError => 2,
+        }
+    }
+}
+
+impl<C, Block> StakingApi<<Block as BlockT>::Hash> for Staking<C, Block>
+where
+    Block: BlockT,
+    C: Send + Sync + 'static,
+    C: ProvideRuntimeApi,
+    C: HeaderBackend<Block>,
+    C::Api: StakingRuntimeApi<Block>,
+{
+    fn get_curve(&self, at: Option<<Block as BlockT>::Hash>) -> Result<Vec<(Perbill, Perbill)>> {
```

Comment:
@vkandy If the RewardCurve points are fixed i.e (Unchangeable as the block grows) then you can remove the block hash part.

Reply:
The block is required to query runtime _as of_ a certain block.  Currently the RPC assumes the best block.

Reply:
We can change that if block is not needed, I think the authority module has some RPC without block hash.